### PR TITLE
[DUOS-3048][risk=no] Populate Study Data Custodians From Dataset Properties

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-3048

### Summary
* This PR adds an admin-only API that allows for updating a study's data custodians field.
* Future manual work will be necessary to make the updates necessary for production studies that are missing data custodians

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java [hunk 1]
```diff
@@ -13,6 +14,8 @@
 import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
```
Comment (aarohinadkarni) [inline src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java]: nit: this import is unused

File: src/main/resources/assets/paths/studyByIdCustodians.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      description: The Updated Study
+      content:
+        application/json:
+          schema:
+            $ref: '../schemas/Study.yaml'
+    400:
+      description: |
+        Bad Request when schema is invalid or the required custodian values are not valid email
+        addresses
+    404:
```
Comment (aarohinadkarni) [inline src/main/resources/assets/paths/studyByIdCustodians.yaml:35]: Should we add 403 for the user not being authorized?

**Review by aarohinadkarni — APPROVED** [review]
thanks for the changes, looks good! 👍

**Review by fboulnois — APPROVED** [review]
looks good 👍

**Pull request merged**